Update roe_records.lua

Daywhite83 opened the pull request:
added roe quests for escha areas

<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [X] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
Adds in the RoE's for Regions Escha - Zi'Tah, Escha - Ru'Aun, and Reisenjima
<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->
All records are working except for Conflict: Escha - Zi'Tah VI, for some reason it does not want to work.

## Steps to test these changes

Step1: zone into escha - zi'tah / ru'aun / resisenjima
Step2: activate the RoE Quests (menu, menu, quests, objective list, Combat (Region), Escha 1 or Escha 2, Activate desired record.
Step3: kill objective mob 10x
Step4: Get reward (exp, sparks, accolades, and either a bead pouch or silt pouch), neither silt or eschan beads are implemented at this time but you can mass store the pouches.
<!-- Clear and detailed steps to test your changes here -->

Daywhite83 committed:
Update roe_records.lua

added roe quests for escha areas

claywar commented on scripts/globals/roe_records.lua:
```diff
@@ -4971,10 +4971,262 @@ function getRoeRecords(triggers)
+        { -- Conflict: Reisenjima X
+            trigger = triggers.mobKill,
+            goal = 10,
+            reqs = { mobName = set { "Glowering Ladybug" }, zone = set { 291 } },
```
Just looking through here, it doesn't appear that we're doing any sort of string manipulation on mob name, so I would assume it should follow the mob_pools sql entry (Underscore instead of space)
`INSERT INTO `mob_pools` VALUES (5387,'Glowering_Ladybug','Glowering_Ladybug',170,0x0000E20700000000000000000000000000000000,6,6,7,220,100,0,1,0,0,0,0,0,1,129,0,0,0,0,0,170,170);`

TeoTwawki replied:
we normally use the name field on mob_spawn_points (tho this should be the same thing as in pools anyway! so technicality) unless an entity is instanced, at which point it matters if it's a battlefield helper (2nd field of mob_pools) or a scripted dynamic entity (defined in script).

TeoTwawki replied:
While we're here...It's beyond the scope of this PR but I'd like to again voice my irritation that we effectively ask every mob in the entire game world its name at this point. Every rock lizard, tunnel worm, crab and behemoth has its name checked against.

Daywhite83 committed:
Update roe_records.lua

replaced spaces with _'s in mob names

Daywhite83 reviewed (commented):
replaced spaces with underscores as requested

Daywhite83 committed:
Update roe_records.lua

the previous numbers for the records were already is use elsewhere. assigned 750-777

Daywhite83 reviewed (commented):
record numbers i was using were already is use elsewhere. Using 750-777 for echa records (hopefully no more conflicts)

claywar commented:
> record numbers i was using were already is use elsewhere. Using 750-777 for echa records (hopefully no more conflicts)

The record numbers are _not_ arbitrary, and map directly.  Please ensure that you have tested this code that they have the appropriate value.

Daywhite83 commented:
How do i find out that information?

claywar commented:
![image](https://user-images.githubusercontent.com/81713309/187528762-19f416b8-b06e-4754-814c-a4a51a5738ef.png)
![image](https://user-images.githubusercontent.com/81713309/187528811-d86eb74a-08d8-4253-a680-61cd4324fcca.png)
![image](https://user-images.githubusercontent.com/81713309/187528875-bec95b9e-9100-4435-920d-37128c1253da.png)

Daywhite83 committed:
Update roe_records.lua

Correct record IDs now

Daywhite83 reviewed (commented):
Has the correct Record #'s now

Daywhite83 commented:
thanks claywar

claywar commented:
Perfect.  Now all you need to do is test that your changes are working.

Daywhite83 committed:
Update roe_records.lua

Daywhite83 reviewed (commented):
for some reason it was showing the old record numbers when i came back to the file, showing correctly now, now to figure out how to test it

CatsEyeXI commented:
@Daywhite83 I have a small group interested in helping you test these out. Would you happen to know what the pre-requisites are to get the quests to appear in the log? 

![image](https://user-images.githubusercontent.com/65316697/187586214-186d809e-2287-4564-b131-856f371ee81c.png)

Daywhite83 commented:
im pretty sure you can get them as soon as you zone into escha the 1st time

CatsEyeXI commented:
> im pretty sure you can get them as soon as you zone into escha the 1st time

Yeah, that's what we were under the impression of as well. Anything needed other than the global lua?

Daywhite83 commented:
nope

RAIST5150 commented:
Small detail... spelling correction needed for Escha Ru'aun objectives:

Il'aern, not Li'aern.  (il, not li)

Daywhite83 committed:
Update roe_records.lua

changed Li'Aern -> Il'Aern
bgwiki did me wrong on that one lol....

EpicTaru reviewed (commented):
Line 5108 has empty spaces that need removed, looks like un-needed tab:

![image](https://user-images.githubusercontent.com/26195580/187802010-da5502f6-cc99-4abc-92cb-d1b9be064b9d.png)

Daywhite83 committed:
Update roe_records.lua

removed whitespace

Daywhite83 reviewed (commented):
Removed Whitespace.

CatsEyeXI commented:
well, I've tried this on a fresh LSB pull, content restriction off, even completed RoV 1-4 (Set Free), and lists are still empty. 
if you think of anything, let me know!
![image](https://user-images.githubusercontent.com/65316697/187819362-12afb147-93c7-4422-9871-d41f4e278450.png)
![image](https://user-images.githubusercontent.com/65316697/187820666-9b752dff-8e4d-4e6a-9c07-c3f8b88f3e48.png)

Daywhite83 commented:
i dont understand why its not working, someone on the discord for the server im on suggests
"maybe zones aren't getting ticked as SaveZonesVisited"
im not sure how to find out whether or not they are or aren't, still very new to all this lol...


Sent from Outlook<http://aka.ms/weboutlook>

________________________________
From: CatsEyeXI ***@***.***>
Sent: August 31, 2022 10:41 PM
To: LandSandBoat/server ***@***.***>
Cc: Daywhite83 ***@***.***>; Mention ***@***.***>
Subject: Re: [LandSandBoat/server] Update roe_records.lua (PR #2624)


well, I've tried this on a fresh LSB pull, content restriction off, even completed RoV 1-4 (Set Free), and lists are still empty.
if you think of anything, let me know!
[image]<https://user-images.githubusercontent.com/65316697/187819362-12afb147-93c7-4422-9871-d41f4e278450.png>
[image]<https://user-images.githubusercontent.com/65316697/187820666-9b752dff-8e4d-4e6a-9c07-c3f8b88f3e48.png>

—
Reply to this email directly, view it on GitHub<https://github.com/LandSandBoat/server/pull/2624#issuecomment-1233667250>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A2NHF6FSWMQ3YPKYVNJNM73V4AJVVANCNFSM6AAAAAAQAUAP7E>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

neuromancerxi commented:
> well, I've tried this on a fresh LSB pull, content restriction off, even completed RoV 1-4 (Set Free), and lists are still empty. if you think of anything, let me know! 

#2627 might explain this behavior.

Daywhite83 commented:
oh yea, it must have been you that mentioned it on xiweb discord lol

EpicTaru commented:
Didn't mean to approve the PR request in entirety, only meant to review the changes made for the review I had left.

CatsEyeXI commented:
> > well, I've tried this on a fresh LSB pull, content restriction off, even completed RoV 1-4 (Set Free), and lists are still empty. if you think of anything, let me know!
> 
> #2627 might explain this behavior.

nailed it.

![image](https://user-images.githubusercontent.com/65316697/187971357-2f4db7b7-2241-4c89-8e24-f1f926fbc834.png)

Daywhite83 commented:
Omg that's awesome

Sent from my Bell Samsung device over Canada’s largest network.
________________________________
From: CatsEyeXI ***@***.***>
Sent: Thursday, September 1, 2022 1:02:37 PM
To: LandSandBoat/server ***@***.***>
Cc: Daywhite83 ***@***.***>; Mention ***@***.***>
Subject: Re: [LandSandBoat/server] Update roe_records.lua (PR #2624)


well, I've tried this on a fresh LSB pull, content restriction off, even completed RoV 1-4 (Set Free), and lists are still empty. if you think of anything, let me know!

#2627<https://github.com/LandSandBoat/server/issues/2627> might explain this behavior.

nailed it.

[image]<https://user-images.githubusercontent.com/65316697/187971357-2f4db7b7-2241-4c89-8e24-f1f926fbc834.png>

—
Reply to this email directly, view it on GitHub<https://github.com/LandSandBoat/server/pull/2624#issuecomment-1234547839>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A2NHF6DE3MHCDKL57GFWY6LV4DOS3ANCNFSM6AAAAAAQAUAP7E>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Daywhite83 commented:
> > > well, I've tried this on a fresh LSB pull, content restriction off, even completed RoV 1-4 (Set Free), and lists are still empty. if you think of anything, let me know!
> > 
> > 
> > #2627 might explain this behavior.
> 
> nailed it.
> 
> ![image](https://user-images.githubusercontent.com/65316697/187971357-2f4db7b7-2241-4c89-8e24-f1f926fbc834.png)

I couldnt replicate your results unfortunately.

CatsEyeXI commented:
Make sure you're on the latest commit.

Daywhite83 commented:
I was able to finally get the quests to pop.
Still not working tho for me, when I kill the mod they don't get checked off in the quest, going to progress through story a bit and see if it's that, or the fact I was killing with godmode on lol....

Daywhite83 commented:
UPDATE: The ones in Reisenjima seem to be giving credit for the kills. Will be doing some testing tomorrow to see if its giving the correct rewards

Daywhite83 commented:
Not working :
Conflict: Escha - Ru'Aun I
Conflict: Escha - Ru'Aun III
Conflict: Escha - Ru'Aun IV
Conflict: Escha - Ru'Aun VI

Working:
Conflict: Escha - Ru'Aun II
Conflict: Escha - Ru'Aun V
Conflict: Escha - Ru'Aun VII
Conflict: Escha - Ru'Aun VIII
Conflict: Escha - Ru'Aun IX
Conflict: Escha - Ru'Aun X

Testing Escha - Zi'Tah next will post Screenshots soon of each quest completing

Daywhite83 commented:
Not working Reisenjima:
Conflict: Reisenjima IV
Conflict: Reisenjima V

Working Reisenjima:
Conflict: Reisenjima I
Conflict: Reisenjima II
Conflict: Reisenjima III
Conflict: Reisenjima VI
Conflict: Reisenjima VII
Conflict: Reisenjima VIII
Conflict: Reisenjima IX
Conflict: Reisenjima X
Conflict: Reisenjima XI

Daywhite83 committed:
Update roe_records.lua

fixed mob naming issues, all records work EXCEPT Escha-Zi'Tah VI (Eschan_Puk) and Escha - Ru'Aun I (Eschan_Il'Aern)

Daywhite83 reviewed (commented):
fixed mob naming issues, all records work EXCEPT Escha-Zi'Tah VI (Eschan_Puk) and Escha - Ru'Aun I (Eschan_Il'Aern)

Daywhite83 committed:
Update roe_records.lua

got escha ru'aun I working, just need to get escha zi'tah 6 working now

Daywhite83 reviewed (commented):
got escha ru'aun I working, just need to get escha zi'tah 6 working now

RAIST5150 commented:
> Just an't seem to get Conflict: Escha - Zi'Tah VI working.... even tried.....
> 
> ```
>     [901] =
>     { -- Conflict: Escha - Zi'Tah VI
>         goal = 10,
>         reqs = { mobID = set { 	17957121, 17957122, 17957123, 17957124, 17957125, 17957128,
> 		17957129, 17957130, 17957131, 17957132, 17957137, 17957138,
> 		17957139, 17957140, 17957141, 17957145, 17957146, 17957147,
> 		17957148, 17957149, 17957152, 17957153, 17957154, 17957155,
> 		17957159, 17957160, 17957161, 17957162}, zone = set { 288 } },
>         flags = set { "repeat" },
>         reward = { sparks = 300, xp = 900, capacity = 150, accolades = 30, item = { 6392 } },
>     },
> ```
> 
> ... with no success, all others work fine now though, i can just comment out #901 if requested.

 trigger = triggers.mobKill,

Needs to know what action to monitor as well as for what mob?

Daywhite83 commented:
Oh my gosh, ok once I get the kids ready for school I'll update test then update the code here and share my success, thanks for noticing that raist

Daywhite83 committed:
Update roe_records.lua

added 
trigger = triggers.mobKill,
            
to line 5020, which should have fixed the issue but still getting no credit,

Daywhite83 commented:
I can confirm Escha -m Zi'Tah VI now works.
![ffxi_2022 09 08_07 44 52](https://user-images.githubusercontent.com/110785272/189114284-d4c5f83a-64c1-4441-bd59-f98128fce679.jpg)

WinterSolstice8 commented:
What's the status on this, is it fully tested?

Daywhite83 commented:
yup its been ready since sept 8
________________________________
From: WinterSolstice8 ***@***.***>
Sent: September 23, 2022 9:19 PM
To: LandSandBoat/server ***@***.***>
Cc: Daywhite83 ***@***.***>; Mention ***@***.***>
Subject: Re: [LandSandBoat/server] Update roe_records.lua (PR #2624)


What's the status on this, is it fully tested?

—
Reply to this email directly, view it on GitHub<https://github.com/LandSandBoat/server/pull/2624#issuecomment-1256825939>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A2NHF6CQEG25T2SAYT44EWLV7ZJI7ANCNFSM6AAAAAAQAUAP7E>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Daywhite83 commented:
I'm still very new to all this, couldn't get the quests to pop for me

Message ID: ***@***.***>
